lib/main.js

```diff
@@ -114,7 +114,9 @@ Lambda.prototype._params = function (program, buffer) {
       Variables: {}
     },
     DeadLetterConfig: {
-      TargetArn: ''
+      // Set default value to null.
+      // Because createFunction fails if it is empty character.
+      TargetArn: null
     }
```

Comment:
It might make more sense to leave it out completely and only add it when passed through to us? This might also solve for your TODO.